runtime/lua/vim/diagnostic.lua

```diff
@@ -59,8 +59,22 @@ local bufnr_and_namespace_cacher_mt = {
   end,
 }
 
-local diagnostic_cleanup = setmetatable({}, bufnr_and_namespace_cacher_mt)
-local diagnostic_cache = setmetatable({}, bufnr_and_namespace_cacher_mt)
+local diagnostic_cache = setmetatable({}, {
+  __index = function(t, bufnr)
+    if not bufnr or bufnr == 0 then
+      bufnr = vim.api.nvim_get_current_buf()
+    end
+
+    vim.api.nvim_buf_attach(bufnr, false, {
+      on_detach = function(_, b)
```

Comment:
I don’t think we need to capture the `b` argument here
```suggestion
      on_detach = function()
```